recipes/aws-cdi-sdk/all/conanfile.py

```diff
@@ -0,0 +1,110 @@
+        if not self.options["aws-libfabric"].shared or not self.options["aws-sdk-cpp"].shared:
+            raise ConanInvalidConfiguration("Cannot build with static dependencies")
+        if not getattr(self.options["aws-sdk-cpp"], "monitoring"):
+            raise ConanInvalidConfiguration("This package requires the monitoring AWS SDK")
+        tools.check_min_cppstd(self, 11)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self)
+        return self._autotools
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.configure()
+        return self._cmake
+
+    def _detect_compilers(self):
+        cmake_cache = tools.load(os.path.join(self.build_folder, "CMakeCache.txt"))
+        cc = re.search("CMAKE_C_COMPILER:FILEPATH=(.*)", cmake_cache)[1]
+        cxx = re.search("CMAKE_CXX_COMPILER:FILEPATH=(.*)", cmake_cache)[1]
+        return cc, cxx
+
+    def build(self):        
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+        # build aws-cpp-sdk-cdi
+        cmake = self._configure_cmake()
+        cmake.build()
+
+        autotools = self._configure_autotools()
+        with tools.chdir(self._source_subfolder):
+            # configure autotools to find aws-cpp-sdk-cdi
+            autotools.include_paths.append(os.path.join(self.build_folder, self._source_subfolder, "aws-cpp-sdk-cdi", "include"))
+            autotools.library_paths.append(os.path.join(self.build_folder, "lib"))
+            autotools.libs.append("aws-cpp-sdk-cdi")
+
+            vars = autotools.vars
+            cc, cxx = self._detect_compilers()
+            vars["CC"] = cc
+            vars["CXX"] = cxx
+            if self.settings.build_type == "Debug":
+                vars["DEBUG"] = "y"
+
+            args = ["require_aws_sdk=no"]
+
+            autotools.make(target="libsdk", vars=vars, args=args)
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+        self.copy(pattern="*", dst="include", src=os.path.join(self._source_subfolder, "include"))
+        config = "debug" if self.settings.build_type == "Debug" else "release"
+        self.copy(pattern="*", dst="lib", src=os.path.join(self._source_subfolder, "build", config, "lib"))
+
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        self.cpp_info.libs = self.collect_libs()
+        if self.settings.os == "Linux":
```

Comment:
It hard to find but https://github.com/aws/aws-sdk-cpp/blob/bd00fe8d76e2a774c6342f659b49d4458658f4c3/cmake/utilities.cmake#L34 is the magic how they do things

The v1 generators need more love 

```suggestion
        self.cpp_info.set_property("cmake_file_name", "aws-cpp-sdk-cdi")
        self.cpp_info.set_property("cmake_target_name", "aws-cpp-sdk-cdi")

        # TODO: to remove in conan v2 once cmake_find_package_* generators removed
        # TODO: Remove the namespace on CMake targets
        self.cpp_info.names["cmake_find_package"] = "aws-cpp-sdk-cdi"
        self.cpp_info.names["cmake_find_package_multi"] = "aws-cpp-sdk-cdi"
        if self.settings.os == "Linux":
```

Reply:
I've split it to two components, one for the C++ SDK named `aws-cpp-sdk-cdi` and one for the C SDK named `cdisdk` like the library.

Reply:
We still need to set the CMake and pkg config files names 🙏 pretty please

Reply:
done